File: backend/models/tag.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type Tag struct {
	ID uint `gorm:"primaryKey" json:"id"`

	Name string `gorm:"size:100;uniqueIndex" json:"name"`

	Slug string `gorm:"size:150;uniqueIndex" json:"slug"`

	Description string `gorm:"type:text" json:"description"`

	Status bool `gorm:"default:true" json:"status"`

	SortOrder int `gorm:"default:0" json:"sort_order"`

	CreatedAt time.Time `json:"created_at"`

	UpdatedAt time.Time `json:"updated_at"`

	DeletedAt gorm.DeletedAt `gorm:"index" json:"-"`
}

File: admin/types/tag.ts
export interface Tag {
    id: number;
    name: string;
    slug: string;
    description: string;
    status: boolean;
    sort_order: number;
    created_at: string;
}

export interface TagResponse {
    success: boolean;
    message: string;
    data: Tag[];
    meta: {
        page: number;
        limit: number;
        total: number;
    };
}